File: snake_qt5/HighestScoreScene.cpp
#include "HighestScoreScene.h"
#include "Config.h"
#include "Director.h"
#include "StartScene.h"
#include "QLabel"
#include "QPushButton"
#include "QGridLayout"

HighestScoreScene::HighestScoreScene(Window *parent)
	: Scene(parent)
{
}

HighestScoreScene::~HighestScoreScene()
{
}

void HighestScoreScene::init()
{
	//控件
	QLabel *infoLabel = new QLabel(this);
	infoLabel->setAlignment(Qt::AlignCenter);
	infoLabel->setFont(QFont(u8"微软雅黑", 40));
	infoLabel->setText(tr(u8"最  高  分"));

	QLabel *highestScoreLabel = new QLabel(this);
	highestScoreLabel->setAlignment(Qt::AlignCenter);
	highestScoreLabel->setFont(QFont(u8"微软雅黑", 25));
	highestScoreLabel->setText(Config::getInstance()->getHighestScorePlayer() + "          " + QString::number(Config::getInstance()->getHighestScore()) + tr(u8"分"));

	QPushButton *returnToStartSceneButton = new QPushButton(this);
	returnToStartSceneButton->setFont(QFont(u8"微软雅黑", 15));
	returnToStartSceneButton->setText(tr(u8"返回开始界面"));
	connect(returnToStartSceneButton, &QPushButton::clicked, this, &HighestScoreScene::returnToStartSceneButtonClicked);

	//布局
	QGridLayout *layout = new QGridLayout(this);
	layout->addWidget(infoLabel, 0, 0, 10, 15);
	layout->addWidget(highestScoreLabel, 10, 5, 3, 5);
	layout->addWidget(returnToStartSceneButton, 13, 5, 3, 5);
}

void HighestScoreScene::returnToStartSceneButtonClicked()
{
	StartScene *startScene = new StartScene(Director::getInstance()->getWindow());
	Director::getInstance()->setNowScene(startScene);
	startScene->init();
	startScene->show();
	deleteLater();
}

File: snake_pyqt5/Scene/HighestScoreScene.py
from Scene.Scene import Scene
from Director import Director
from Config import Config
from PyQt5.QtWidgets import QLabel, QPushButton, QGridLayout
from PyQt5.QtGui import QFont
from PyQt5.QtCore import Qt


#最高分场景
class HighestScoreScene(Scene):
    def __init__(self, parentWindow):
        super().__init__(parentWindow)

    #初始化场景
    def init(self):
        #控件
        infoLabel = QLabel(self)
        infoLabel.setAlignment(Qt.AlignCenter)
        infoLabel.setFont(QFont("微软雅黑", 40))
        infoLabel.setText(self.tr("最  高  分"))

        highestScoreLabel = QLabel(self)
        highestScoreLabel.setAlignment(Qt.AlignCenter)
        highestScoreLabel.setFont(QFont("微软雅黑", 25))
        highestScoreLabel.setText("%d%s      %s" %
                                  (Config().highestScore, self.tr("分"),
                                   Config().highestScorePlayer))

        returnToStartSceneButton = QPushButton(self)
        returnToStartSceneButton.setFont(QFont("微软雅黑", 15))
        returnToStartSceneButton.setText(self.tr("返回开始界面"))
        returnToStartSceneButton.clicked.connect(
            self.returnToStartSceneButtonClicked)

        #布局
        layout = QGridLayout(self)
        layout.addWidget(infoLabel, 0, 0, 10, 15)
        layout.addWidget(highestScoreLabel, 10, 5, 3, 5)
        layout.addWidget(returnToStartSceneButton, 13, 5, 3, 5)

    #单击返回开始场景按钮
    def returnToStartSceneButtonClicked(self):
        from Scene.StartScene import StartScene
        startScene = StartScene(Director().window)
        Director().nowScene = startScene
        startScene.init()
        startScene.show()
        self.deleteLater()
